l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -634,51 +649,52 @@ def setDettaglioLinea(
-            Quantita=('%.' + str(
-                uom_precision
-            ) + 'f') % line.quantity,
-            UnitaMisura=line.uos_id and (
-                unidecode(line.uos_id.name)) or None,
-            PrezzoTotale='%.2f' % line.price_subtotal,
+            PrezzoUnitario=('%.' + str(price_precision) + 'f') % float_round(
+                prezzo_unitario, price_precision),
+            Quantita=('%.' + str(uom_precision) + 'f') % line.quantity,
+            UnitaMisura=line.uos_id and (unidecode(line.uos_id.name)) or None,
+            PrezzoTotale='%.2f' % float_round(line.price_subtotal, price_precision),
             AliquotaIVA=AliquotaIVA)
-        if line.discount:
-            ScontoMaggiorazione = ScontoMaggiorazioneType(
-                Tipo='SC',
-                Percentuale='%.2f' % line.discount
-            )
-            DettaglioLinea.ScontoMaggiorazione.append(ScontoMaggiorazione)
+        DettaglioLinea.ScontoMaggiorazione.extend(
+            self.setScontoMaggiorazione(line))
         if aliquota == 0.0:
             if not line.invoice_line_tax_id[0].kind_id:
                 raise UserError(
-                    _("No 'nature' field for tax %s") %
+                    _("No 'nature' field for tax %s.") %
                     line.invoice_line_tax_id[0].name)
             DettaglioLinea.Natura = line.invoice_line_tax_id[
                 0
             ].kind_id.code
         if line.admin_ref:
             DettaglioLinea.RiferimentoAmministrazione = line.admin_ref
         if line.product_id:
-            if line.product_id.default_code:
+            product_code = line.product_id.default_code
+            if product_code:
                 CodiceArticolo = CodiceArticoloType(
                     CodiceTipo=self.env['ir.config_parameter'].sudo(
                     ).get_param('fatturapa.codicetipo.odoo', 'ODOO'),
-                    CodiceValore=encode_for_export(
-                        line.product_id.default_code, 35, 'ascii')
+                    CodiceValore=product_code[:35],
                 )
                 DettaglioLinea.CodiceArticolo.append(CodiceArticolo)
-            if line.product_id.ean13:
+            product_barcode = line.product_id.ean13
+            if product_barcode:
                 CodiceArticolo = CodiceArticoloType(
                     CodiceTipo='EAN',
-                    CodiceValore=encode_for_export(
-                        line.product_id.ean13, 35, 'ascii')
+                    CodiceValore=product_barcode[:35],
                 )
                 DettaglioLinea.CodiceArticolo.append(CodiceArticolo)
 
         body.DatiBeniServizi.DettaglioLinee.append(DettaglioLinea)
-        return True
+        return DettaglioLinea
+
+    def setScontoMaggiorazione(self, line):
+        res = []
+        if line.discount:
+            res.append(ScontoMaggiorazioneType(
+                Tipo='SC',
+                Percentuale='%.2f' % float_round(line.discount, 8)
```

Comment:
```suggestion
                Percentuale='%.8f' % float_round(line.discount, 8)
```
Altrimenti i decimali dopo i primi 2 vanno persi no?

Reply:
Sono d'accordo con te sul fatto che non ha senso il round a 8 se poi si castra a 2. 
Questo problema sta anche sulla v12, ho lasciato la stessa logica per uniformità...

https://github.com/OCA/l10n-italy/blob/ee36257004b52c185123bdd27f0633aaeb5d2593/l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py#L697-L704

Reply:
inoltre sollevai le stesse perplessità, senza però suggerire nessuna correzione quando si fece la castrazione a 2:

https://github.com/OCA/l10n-italy/pull/1873

Reply:
vero